### DISCUSSION TITLE
How to send status code along with trailers in a Body::channel?

### DISCUSSION THREAD

**Discussion by ramiroaisen** [hyperium/hyper#3005] [Q&A]
Hello, first of all, thanks for this great project!

I'm using hyper to create an Icecast compatible streaming server and I didn't find a way to send a status code along with trailers in a body constructed from `Body::channel`.

Is this possible? I only see that `send_trailers` accepts a `HeaderMap` but no `StatusCode`.

I'm sure I'm missing something.

Thank you!

**Comment by seanmonstar** [comment]
I'm not familiar with Icecast (it looks like it's just HTTP), but HTTP doesn't have a concept of sending an HTTP status code in the trailers. That's a field that is returned in an `http::Response`. It might be that Icecast puts a specific trailer with a code, if so, you could just do `headers.insert("status-whatever", my_value)`.

**Reply by ramiroaisen** [reply]
Yes, it's based in http 1.0 with some odds, I'm following the spec described here: https://gist.github.com/ePirat/adc3b8ba00d85b7e3870 (its a one page document)
In there it specify a 100 Continue first status and then a 200 OK after the data is received.

**Reply by seanmonstar** [reply]
Oh, it's sending an `Expect` header. hyper will automatically send the 100 continue when you try to read from the body. You just need to eventually return a response with a 200 once done.
